Model/ship.php

```diff
@@ -22,7 +22,7 @@ public function getType()
     public function isFunctional()
     {
         // check that the currentHealth is >= 15% of the maxHealth
-        if (parent::getCurrentHealth() <= ((20/100) * parent::getMaxHealth())) {
+        if ($this->getCurrentHealth() <= ((20/100) * $this->getMaxHealth())) {
```

Comment:
You used $this, but you didn't update the logic.

Reply:
Fixed the logic to be simplified